src/evo/mnauth.cpp

```diff
@@ -25,6 +26,21 @@ void CMNAuth::PushMNAUTH(CNode& peer, CConnman& connman, const CBlockIndex* tip)
         return;
     }
 
+    // NOTE: this relies on the fact that:
+    // - regtest masternodes are all 127.0.0.1,
```

Comment:
maybe you can create functional test for that based on fact that 127.0.0.* all equivalent, so, you can use `127.0.0.33` and actually receive message, but it would be _different_ IP. Is it possible?